src/erlcloud_cloudsearch.erl

```diff
@@ -0,0 +1,675 @@
+    add_tags(DomainARN, TagList, default_config()).
+
+-spec add_tags(DomainName :: string(),
+               TagList :: [cloudsearch_domain_tag()],
+               Config :: aws_config()) -> cloudsearch_return_val().
+add_tags(DomainARN, TagList, Config) ->
+    TagParams = get_tag_params(TagList),
+    cloudsearch_query(Config, "AddTags", [{"ARN", DomainARN} | TagParams]).
+
+%%------------------------------------------------------------------------------
+%% CreateDomain
+%%
+%% http://docs.aws.amazon.com/cloudsearch/latest/developerguide/API_CreateDomain.html
+%%------------------------------------------------------------------------------
+
+-spec create_domain(DomainName :: string()) -> cloudsearch_return_val().
+create_domain(DomainName) ->
+    create_domain(DomainName, default_config()).
+
+-spec create_domain(DomainName :: string(), Config :: aws_config()) ->
+    cloudsearch_return_val().
+create_domain(DomainName, Config) ->
+    cloudsearch_query(Config, "CreateDomain", [{"DomainName", DomainName}]).
+
+%%------------------------------------------------------------------------------
+%% DefineIndexField
+%%
+%% http://docs.aws.amazon.com/cloudsearch/latest/developerguide/API_DefineIndexField.html
+%%
+%% `define_index_field/2` and `define_index_field/3` functions allow to configure
+%% single field for the search domain.
+%% Whereas `define_index_fields/2` and `define_index_fields/3` allow to configure
+%% multiple fields at a time. Check AWS documentation for details.
+%%
+%% Example (multiple fields in single call):
+%% IndexFields = [erlcloud_cloudsearch:int_field_options("field1"),
+%%                erlcloud_cloudsearch:int_array_field_options("field2"),
+%%                erlcloud_cloudsearch:text_field_options("field3")].
+%% {ok, Response} = define_index_fields("my-test-domain", IndexFields).
+%%------------------------------------------------------------------------------
+
+-spec define_index_field(DomainName :: string(), IndexField :: proplist()) ->
+    cloudsearch_return_val().
+define_index_field(DomainName, IndexField) ->
+    define_index_field(DomainName, IndexField, default_config()).
+
+-spec define_index_field(DomainName :: string(),
+                         IndexField :: proplist(),
+                         Config :: aws_config()) ->
+    cloudsearch_return_val().
+define_index_field(DomainName, IndexField, Config) ->
+    define_index_fields(DomainName, [IndexField], Config).
+
+-spec define_index_fields(DomainName :: string(),  IndexFields :: [proplist()]) ->
+    cloudsearch_return_val().
+define_index_fields(DomainName, [[{_,_}|_]|_] = IndexFields) ->
+    define_index_fields(DomainName, IndexFields, default_config()).
+
+-spec define_index_fields(DomainName :: string(),
+                          IndexFields :: [proplist()],
+                          Config :: aws_config()) ->
+    cloudsearch_return_val().
+define_index_fields(DomainName, [[{_,_}|_]|_] = IndexFields, Config) ->
```

Comment:
@kingbrewer I would say that adding a pattern match of `[[{_,_}|_]|_]` actually tells clients very little about the input that is expected, and I would urge you to consider using named types if you want to be a good module author who is helpful to his users. Your type spec of `[proplist()]` is just as useful as this is, IMO. You could do a lot better, but I don't know that it's strictly necessary. The example appears to cover it reasonably well.

As for the config guard, I don't really care either way. The spec says what it should be, and if clients violate it, it will surely crash one way or the other. Wouldn't hurt to do it, but I don't think I've seen a consistent standard for that in the library.